sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ImmutableStorageWithVersioningTests.java

```diff
@@ -94,7 +94,7 @@ public class ImmutableStorageWithVersioningTests extends BlobTestBase {
     private static final String RESOURCE_GROUP_NAME = ENVIRONMENT.getResourceGroupName();
     private static final String SUBSCRIPTION_ID = ENVIRONMENT.getSubscriptionId();
     private static final String API_VERSION = "2021-04-01";
-    private static final TokenCredential CREDENTIAL = new EnvironmentCredentialBuilder().build();
+    private static final TokenCredential CREDENTIAL = new DefaultAzureCredentialBuilder().build();
```

Comment:
When running in Live mode, avoid using DefaultAzureCredentialBuilder directlty.
 
Use this logic: 
```java
public class TestUtil {

    /**
     * Gets a token credential for use in tests.
     * @param interceptorManager the interceptor manager
     * @return the TokenCredential
     */
    public static TokenCredential getTestTokenCredential(InterceptorManager interceptorManager) {
        if (interceptorManager.isLiveMode()) {
            return new AzurePowerShellCredentialBuilder().build();
        } else if (interceptorManager.isRecordMode()) {
            return new DefaultAzureCredentialBuilder().build();
        } else {
            return getPlaybackTokenCredential();
        }
    }

    /**
     * Gets a playback token credential for use in tests.
     * @return the TokenCredential
     */
    public static TokenCredential getPlaybackTokenCredential() {
        return new MockTokenCredential();
    }
}
```

Reply:
We're running into the issue of using interceptorManager, interceptorManager is a non-static variable which is why we're unable to grab it within the static method. We tried to use @TestInstance(TestInstance.Lifecycle.PER_CLASS) defined in the test classes, but this doesn't resolve the issue. How should we go about using interceptorManager, considering the fact that it is a non-static variable being initialized in TestBase.class?

Reply:
at the end, its about reading the env variable. 
If the current logic doesn't allow you to work with Interceptor Manager.
You can read the env variable AZURE_TEST_MODE and check if its PLAYBACK, LIVE or RECORD as an alternative.